common/rpc/factory.go

```diff
@@ -18,27 +18,26 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package config
+package rpc
 
 import (
-	"errors"
 	"fmt"
 	"net"
-	"strings"
 	"sync"
 
 	"go.uber.org/yarpc"
 	"go.uber.org/yarpc/api/transport"
 	"go.uber.org/yarpc/transport/grpc"
 	"go.uber.org/yarpc/transport/tchannel"
 
+	"github.com/uber/cadence/common/config"
 	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/log/tag"
 )
 
-// RPCFactory is an implementation of service.RPCFactory interface
-type RPCFactory struct {
-	config      *RPC
+// Factory is an implementation of service.RPCFactory interface
+type Factory struct {
+	config      config.RPC
 	serviceName string
 	ch          *tchannel.ChannelTransport
```

Comment:
nit and not directly related to this change: rename ch to tchannel?

Reply:
Agree, fixed.